superset/views/geocoding.py

```diff
@@ -0,0 +1,327 @@
+            return self.coder.geocode("MapTiler", data)
+
+    def _add_lat_lon_columns(self, table_name: str, lat_column: str, lon_column: str):
+        """
+        Add new longitued and latitude columns to table
+        :param table_name: The table name of table to insert columns
+        :param lat_column: The name of latitude column
+        :param lon_column: The name of longitude column
+        :raise SqlAddColumnException: When add column-SQL went wrong
+        """
+        connection = db.engine.connect()
+        transaction = connection.begin()
+        try:
+            self._add_column(connection, table_name, lat_column, Float())
+            self._add_column(connection, table_name, lon_column, Float())
+            transaction.commit()
+        except Exception:
+            transaction.rollback()
+            raise SqlAddColumnException(
+                "An error occured while creating new columns for latitude and longitude"
+            )
+
+    def _add_column(
+        self,
+        connection: Connection,
+        table_name: str,
+        column_name: str,
+        column_type: str,
+    ):
+        """
+        Add new column to table
+        :param connection: The connection to work with
+        :param table_name: The name of table to add a new column
+        :param column_name: The name of latitude column
+        :param column_type: The name of longitude column
+        """
+        column = Column(column_name, column_type)
+        name = column.compile(column_name, dialect=db.engine.dialect)
+        type = column.type.compile(db.engine.dialect)
+        sql = text("ALTER TABLE %s ADD %s %s" % (table_name, name, type))
+        connection.execute(sql)
+
+    def _insert_geocoded_data(
+        self,
+        table_name: str,
+        lat_column: str,
+        lon_column: str,
+        geo_columns: list,
+        data: list,
+    ):
+        """
+        Insert geocoded coordinates in table
+        :param table_name: The name of table to insert
+        :param lat_column: The name of latitude column
+        :param lon_column: The name of longitude column
+        :param geo_columns: The list of selected geographical column names
+        :param data: row with geographical data and geocoded coordinates
+        :raise SqlUpdateException: When Update of given columns with given data went wrong
+        """
+        where_clause = "='%s' AND ".join(filter(None, geo_columns)) + "='%s'"
+        number_of_columns = len(geo_columns)
+
+        connection = db.engine.connect()
+        transaction = connection.begin()
+        try:
+            for row in data:
+                update = "UPDATE %s SET %s=%s, %s=%s " % (
+                    table_name,
+                    lat_column,
+                    row[number_of_columns],
+                    lon_column,
+                    row[number_of_columns + 1],
+                )
+                where = "WHERE " + where_clause % (row[:number_of_columns])
```

Comment:
Does this work? SQL-command would look like: "WHERE ='rows' AND columns='values'" or am I reading this wrong?

Reply:
This would look like:
WHERE column_name1 = 'xxx' AND column_name2 = 'yyy'
:number_of_columns  gets the first column names of rows (like address, country etc.) and fill they in the "where_clause"

Reply:
Jeah did not understand the previous statement, now it makes sense.